src/linker/rule.py

```diff
@@ -90,8 +109,9 @@ def _build_shell_command(self) -> str:
         for var_name, var_value in self.envvars.items():
             shell_cmd += f"""
         export {var_name}={var_value}"""
+        # Log stdout/stderr to diagnostics directory
         shell_cmd += f"""
-        {self.script_cmd}
+        {self.script_cmd} > {{log}} 2>&1
```

Comment:
I'll note: we could consider having separate files. I have never been particularly interested in the difference, though, and maybe our users aren't either.

Reply:
Let's leave as-is for now!